deploy_pulp3/pulp3_systemd/tasks/main.yml

```diff
@@ -0,0 +1,30 @@
+- name: Install Systemd unit files
+  # The template allows us to specify the platform python environment.
+  template:
+      src: "templates/{{ item }}.j2"
+      dest: "/etc/systemd/system/{{ item }}.service"
+  with_items:
+    - pulp_celerybeat
+    - pulp_resource_manager
+    - pulp_worker@
+  register: result
+ 
+- name: Create /var/cache/pulp directory
+  file:
+    path: /var/cache/pulp
+    state: directory
+    owner: "{{ pulp_user }}"
+    group: "{{ pulp_user }}"
+
+- name: reload systemd units
+  systemd:
+    daemon-reload: yes
+    name: "{{ item }}"
+    state: restarted
+    enabled: yes
+  with_items:
+    - pulp_celerybeat
+    - pulp_resource_manager
+    - pulp_worker@1
+    - pulp_worker@2
+  when: result|changed
```

Comment:
Since, we are using `result|changed` variable, which is the overall changed status of the 'Install Systemd unit files' task, we would end up restarting each and every service even if just one of them were installed. Shouldn't we loop over the `results` attribute inside `result`, in order to restart only that particular service? A better naming for the registered variable in that case would be something like `services_installed`.
Also, I don't understand why there is `pulp_worker@1` and `pulp_worker@2` here. Maybe, I am missing something.